### DISCUSSION TITLE
Use case idea: alternative to github1s.com

### DISCUSSION THREAD

**Discussion by jtpio** [jupyterlite/jupyterlite#65] [Ideas]
https://github1s.com makes it very easy to browse and read code on GitHub with VS Code, and boots in seconds.

It would be cool to have something similar that would open JupyterLite instead. If the repo on GitHub contains notebooks, they could then be executed and interacted with easily.

Opening this issue so we can keep that in mind when implementing different ways of retrieving and sharing content, and also when thinking about deployments and publishing stories.

**Comment by jtpio** [comment]
Of course Binder already does that too.

But the difference here would be that it would take seconds to boot instead of minutes.

**Comment by bollwyvl** [comment]
Whatever the nuts and bolts of the deployment mechanism are, I feel like
keeping the scope of _this_ repo to "build and test the building blocks"
makes the most sense... But unlocking federated extensions, and leaving the
right hooks open on the stuff here, is going to go really far to enabling
these behaviors.

A quick path to a read-remote-only-edit locally experience, is an nbconvert
exporter that could be gotten into nbviewer... It's already solved the
"download the notebook" part, even if the cache busting is kinda... busted
right now.

The original, lost prototype of the jyve stuff was actually built on
dat/beaker, and I'd love to spin back through the distributed stuff now
that it's had a few years to mature. Having the app be editable from within
itself was kinda magic... Though I'm more bullish on the protocol than the
browser.

What actually brought me here was a discussion with @yuvipanda, who's been
cooking on the deployment ideas for dinner time. Even a fully interactive
"waiting room" skin for binder would be kinda magical... And could even
offer to switch to the live kernel when ready.
